pkg/controller/bucket/bucket_controller.go

```diff
@@ -143,27 +158,53 @@ func (bl *bucketListener) Delete(ctx context.Context, obj *v1alpha1.Bucket) erro
 	}
 
 	req := osspec.ProvisionerDeleteBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
 	}
 
 	switch obj.Spec.Protocol.Name {
 	case v1alpha1.ProtocolNameS3:
+		req.BucketName = obj.Spec.Protocol.S3.BucketName
 		req.Region = obj.Spec.Protocol.S3.Region
+		req.BucketContext["Version"] = obj.Spec.Protocol.S3.Version
+		req.BucketContext["SignatureVersion"] = string(obj.Spec.Protocol.S3.SignatureVersion)
+		req.BucketContext["Endpoint"] = obj.Spec.Protocol.S3.Endpoint
 	case v1alpha1.ProtocolNameAzure:
+		req.BucketName = obj.Spec.Protocol.AzureBlob.ContainerName
+		req.BucketContext["StorageAccount"] = obj.Spec.Protocol.AzureBlob.StorageAccount
 	case v1alpha1.ProtocolNameGCS:
+		req.BucketName = obj.Spec.Protocol.GCS.BucketName
+		req.BucketContext["ServiceAccount"] = obj.Spec.Protocol.GCS.ServiceAccount
+		req.BucketContext["PrivateKeyName"] = obj.Spec.Protocol.GCS.PrivateKeyName
+		req.BucketContext["ProjectID"] = obj.Spec.Protocol.GCS.ProjectID
```

Comment:
Same as above, how are these values expected from Bucket? Where do the Central Controller get these?